README.md

```diff
@@ -1,16 +1,531 @@
+[astropy.coordinates.SkyCoord]: https://docs.astropy.org/en/stable/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord
+[astropy.units.Quantity]: https://docs.astropy.org/en/stable/api/astropy.units.Quantity.html#astropy.units.Quantity
+[custom column types]: https://docs.sqlalchemy.org/en/14/core/custom_types.html
+[Google Cloud SQL]: https://cloud.google.com/sql
+[Górski et al. (2005)]: https://doi.org/10.1086/427976
+[gravitational-wave probability sky maps]: https://emfollow.docs.ligo.org/userguide/tutorial/skymaps.html
+[GW200115_042309]: https://doi.org/10.3847/2041-8213/ac082e
+[H3C]: http://cdsarc.u-strasbg.fr/h3c
+[HEALPix]: https://healpix.sourceforge.io
+[hierarchical progressive surveys (HiPS)]: https://www.ivoa.net/documents/HiPS/
+[ICRS]: https://docs.astropy.org/en/stable/api/astropy.coordinates.builtin_frames.ICRS.html#astropy.coordinates.builtin_frames.ICRS
+[ligo-segments]: https://lscsoft.docs.ligo.org/ligo-segments/
+[MOCPy documentation]: https://cds-astro.github.io/mocpy/
+[MOCPy]: https://github.com/tboch/mocpy
+[multi-order coverage (MOC)]: https://ivoa.net/documents/MOC/
+[multi-order]: https://doi.org/10.1051/0004-6361/201526549
+[pg_healpix]: https://github.com/segasai/pg_healpix
+[PgSphere]: https://pgsphere.github.io
+[PostGIS]: https://postgis.net
+[PostgreSQL]: https://www.postgresql.org
+[pyranges]: https://github.com/biocore-ntnu/pyranges
+[Q3C]: https://github.com/segasai/q3c
+[range types]: https://www.postgresql.org/docs/current/rangetypes.html
+[S200115j]: https://gracedb.ligo.org/superevents/S200115j/view/
+[Singer & Price (2015)]: https://doi.org/10.1103/PhysRevD.93.024013
+[SQLAlchemy bulk insertion]: https://docs.sqlalchemy.org/en/14/orm/persistence_techniques.html#bulk-operations
+[SQLAlchemy]: https://www.sqlalchemy.org
+[UNIQ]: https://healpix.sourceforge.io/doc/html/intro_Geometric_Algebraic_Propert.htm#SECTION420
+[Zwicky Transient Facility]: https://www.ztf.caltech.edu
+
+## Theory of operation
+
+### HEALPix basics
+
+[HEALPix] is a scheme for subdividing and indexing the unit sphere, originally
+described by [Górski et al. (2005)]. Although it was originally designed for
+cosmic microwave background analysis, it has found many uses in astronomy,
+particularly through [multi-order coverage (MOC)] maps and
+[hierarchical progressive surveys (HiPS)] used extensively in the [Aladin]
+astronomical information system. It is also used by LIGO and Virgo to store and
+communicate [gravitational-wave probability sky maps].
+
+HEALPix can be thought of as a tree. At the lowest resolution, level 0, HEALPix
+subdivides the sphere into 12 equal-area base tiles, assigned integer indicies
+0 through 11. At level 1, each of the 12 base tiles is subdivided into 4 tiles.
+Every subsequent level divides each of the preceding level's tiles into 4 new
+tiles. At a given level, each of the base pixels has been divided into
+_nside_ = 2<sup>_level_</sup> pixels on a side. Thus there are
```

Comment:
This is a very nice README!

I found the "on a side" text a bit confusing: maybe 4**level would be clearer to understand, given the preceding explanation?  Then also it simply becomes npix = 12 * 4**level below.

Reply:
Should be fixed now. I changed it to:

> At a given level, each of the base pixels has been divided into 4<sup>_level_</sup> pixels (_nside_ = 2<sup>_level_</sup> pixels on each side). Thus there are _npix_ = 12×4<sup>_level_</sup> pixels at a given resolution, assigned integer indices from 0 through (_npix_-1).

because _nside_ is a really commonly used way to describe HEALPix resolution.